File: fixtures/scale/java/large/src/cryptobench/ciphers/tea/TeaSuite.java
package cryptobench.ciphers.tea;

import cryptobench.core.Cipher;
import cryptobench.core.CipherSuite;
import cryptobench.core.SuiteResult;
import cryptobench.verify.RoundTrip;

public class TeaSuite implements CipherSuite {
    @Override
    public String name() {
        return "tea";
    }

    @Override
    public SuiteResult run() {
        Cipher cipher = new TeaCipher(TeaKey.defaultKey());
        int passed = 0;
        int failed = 0;
        long start = System.nanoTime();
        for (String sample : TeaVectors.samples()) {
            if (RoundTrip.check(cipher, sample)) passed++; else failed++;
        }
        return new SuiteResult(name(), passed, failed, System.nanoTime() - start);
    }
}

File: fixtures/scale/dart/large/lib/ciphers/tea/tea_suite.dart
import '../../core/cipher.dart';
import '../../core/cipher_suite.dart';
import '../../core/suite_result.dart';
import '../../verify/round_trip.dart';
import 'tea_cipher.dart';
import 'tea_key.dart';
import 'tea_vectors.dart';

class TeaSuite implements CipherSuite {
  @override
  String name() => 'tea';

  @override
  String category() => 'cipher';

  @override
  SuiteResult run() {
    final Cipher cipher = TeaCipher(TeaKey.defaultKey());
    var passed = 0;
    var failed = 0;
    final start = DateTime.now();
    for (final sample in teaSamples()) {
      if (roundTripCheck(cipher, sample)) {
        passed++;
      } else {
        failed++;
      }
    }
    final elapsed = DateTime.now().difference(start).inMicroseconds;
    return SuiteResult(name(), passed, failed, elapsed);
  }
}
